### ISSUE TITLE
Should I not just use the semantic-UI directly?

### ISSUE THREAD

**Issue by soichih** [CroudSupport/vue-semantic#12]
Thanks for this repo. I am very new to vue, and I don't quite understand the benefit of using this library, rather than using semantic-ui element directly. Can you explain what this does? Sorry if this is a stupid question.

**Comment by green-coder** [comment]
IMHO, this is not a stupid question, it is quite pertinent and is hopefully opening a useful discussion amongst the regular and potential users.

IMHO, some Semantic-UI elements could clearly be used directly (e.g. the icons and buttons). And some others (e.g. dropdowns, modals) could benefit from being used via this library, provided that the design of its Vue components fits the user's needs.

IMHO, this library is the most useful as a starting point to help Vue users figure out how to integrate the Semantic-UI's behaviors in their own Vue components.

**Comment by soichih** [comment]
@green-coder Thanks for the comment. So far 90% of semantic-ui components are CSS only and I can just use it inside the template. For dropdown, popup, that needs initialization, all I have to do is

```
  mounted: function() {
    $(this.$el).find('.ui.dropdown').dropdown();
  },
```
I am not sure if there is a better way?

I've had a bit of issue with more complex elements like modal, but I believe it's just a matter of finding a right pattern that should be documented by semantic-ui. In general, I agree with your opinion that Vue can directly use semantic-ui for the most part.

**Comment by annyHu** [comment]
I install  in vue,but i  can't run it ,it's error ,say couldn't find preset 'es2015'  , why???

**Comment by BrockReece** [comment]
Hi @soichih and @green-coder 

Thanks for your question, I agree that most of the semantic-ui library can just be implemented with css inline, but I built a set of reusable components that help out with bootstrapping more complex parts of the library. I moved these components into a plugin as I was using it in multiple projects and I open sourced it just incase anyone found the components useful.

I hope that helps?

@uaresb are you happy to open a new issue with that comment to keep things tidy?
